src/http/client.rs

```diff
@@ -105,23 +104,27 @@ impl EspHttpClient {
             Ok(Self {
                 raw,
                 follow_redirects_policy: configuration.follow_redirects_policy,
+                event_handler,
             })
         }
     }
 
     extern "C" fn on_events(event: *mut esp_http_client_event_t) -> i32 {
-        let event = unsafe { event.as_mut().unwrap() };
-
-        let handler =
-            unsafe { EVENT_HANDLERS.lock(|handlers| handlers.get(&event.client).copied()) };
+        match unsafe { event.as_mut() } {
+            Some(event) => {
+                let event_handler = unsafe {
+                    (event.user_data as *const Option<NonNull<dyn Fn(&esp_http_client_event_t)>>).as_ref()
+                };
+                if let Some(opt_handler) = event_handler {
+                    let handler = unsafe { opt_handler.unwrap().as_mut() };
```

Comment:
If I can believe my code (which uses a similar, but not identical http client implementation), this unwrap might actually panic.

Reply:
I do not understand this statement. Isn't this *your* code?

Reply:
The change in the PR is mine, but it was originally made for a different http client, not the one in this repository. What I mean by my code, means that other client and the application that uses it. Not important, really.